pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -72,12 +72,58 @@ type MongoDBSpec struct {
 	// configuration file: https://docs.mongodb.com/manual/reference/configuration-options/
 	// +kubebuilder:validation:Type=object
 	AdditionalMongodConfig MongodConfiguration `json:"additionalMongodConfig,omitempty"`
+
+	// User-specified custom MongoDB roles that should be configured in the deployment.
+	// +optional
+	Roles []AutomationConfigRole `json:"roles,omitempty"`
```

Comment:
Unfortunately, the `AutomationConfigRole` struct has a counterpart in `automationconfig` called `CustomRole`, which has the same structure. The reason this was necessary is because `operator-sdk generate k8s` only generates structs that are defined in this file, but the same struct is needed in the automation config. Since `mongodb_types` imports `automationconfig`, I can't import `mongodb_types` in `automationconfig` (lest I introduce a circular dependency).

Is there a way to avoid this duplication?